src/vcSHP.cpp

```diff
@@ -0,0 +1,738 @@
+  break;
+  case vcSHPType::vcSHPType_Polyline:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_LineString, record.data.poly.points, record.data.poly.pointsCount);
+  break;
+  case vcSHPType::vcSHPType_Polygon:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_Polygon, record.data.poly.points, record.data.poly.pointsCount);
+  break;
+  case vcSHPType::vcSHPType_MultiPoint:
+  vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_MultiPoint, record.data.multiPointZ.points, record.data.multiPointZ.pointsCount);
+  break;
+  case vcSHPType::vcSHPType_PointZ:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_Point, &record.data.pointZ.point, 1);
+  break;
+  case vcSHPType::vcSHPType_PolylineZ:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_LineString, record.data.polyZ.points, record.data.polyZ.pointsCount);
+  break;
+  case vcSHPType::vcSHPType_PolygonZ:
+  case vcSHPType::vcSHPType_MultiPatch:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_Polygon, record.data.polyZ.points, record.data.polyZ.pointsCount);
+  break;
+  case vcSHPType::vcSHPType_MultiPointZ:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_MultiPoint, record.data.multiPointZ.points, record.data.multiPointZ.pointsCount);
+  break;
+  case vcSHPType::vcSHPType_PointM:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_Point, &record.data.pointM.point, 1);
+  break;
+  case vcSHPType::vcSHPType_PolylineM:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_LineString, record.data.polyM.points, record.data.polyM.pointsCount);
+  break;
+  case vcSHPType::vcSHPType_PolygonM:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_Polygon, record.data.polyM.points, record.data.polyM.pointsCount);
+  break;
+  case vcSHPType::vcSHPType_MultiPointM:
+    vcProject_UpdateNodeGeometryFromCartesian(&pProgramState->activeProject, pNode, pProgramState->geozone, udPGT_MultiPoint, record.data.multiPointM.points, record.data.multiPointM.pointsCount);
+  break;
+  default:
+    printf("invalid shape type %d\n", (int)record.shapeType);
+  break;
+  }
+  
+}
+
+udResult vcSHP_Load(vcState *pProgramState, const char *pFilename)
+{
+  vcSHP shp = {};
+  udProjectNode *pParentNode = nullptr;
+  uint16_t stringIndex = 0;
+  vcDBF_Record *pDBFRecord = nullptr;
+  udGeoZone zone = {};
+
+  udResult result = vcSHP_LoadFileGroup(&shp, pFilename);
+  UD_ERROR_CHECK(result);
```

Comment:
`UD_ERROR_HANDLE()` will do the equivalent of this, however, you should call `vcSHP_LoadFileGroup` in this macro instead.

Reply:
fixed.